src/SessionSetup/ConnectToStadiaWidget.cpp

```diff
@@ -264,12 +264,21 @@ void ConnectToStadiaWidget::ReloadInstances() {
   });
 }
 
-void ConnectToStadiaWidget::CheckCredentialsAvailable() {
+void ConnectToStadiaWidget::CheckCredentialsAvailableOrLoad() {
   CHECK(selected_instance_.has_value());
 
   const std::string instance_id = selected_instance_->id.toStdString();
 
-  if (!instance_credentials_.contains(instance_id)) return;
+  if (!instance_credentials_.contains(instance_id)) {
+    if (!instance_credentials_loading_.contains(instance_id)) {
+      ggp_client_->GetSshInfoAsync(
+          selected_instance_.value(),
+          [this, instance_id = instance_id](outcome::result<orbit_ggp::SshInfo> ssh_info_result) {
```

Comment:
pers pref: a simple `instance_id` instead of `instance_id = instance_id`.

Reply:
Fixed